### PR TITLE
#338: fix review times in QoS

### LINKED ISSUE #338: `NoMethodError: undefined method - for nil` in the `quality-of-service.rb`
I found this [here](https://github.com/zerocracy/judges-action/actions/runs/10884491018/job/30199895449):

```
NoMethodError: undefined method `-' for nil
	/action/judges/quality-of-service/quality-of-service.rb:138:in `block (3 levels) in <top (required)>'
	/action/judges/quality-of-service/quality-of-service.rb:135:in `each'
	/action/judges/quality-of-service/quality-of-service.rb:135:in `block (2 levels) in <top (required)>'
	/action/judges/quality-of-service/quality-of-service.rb:132:in `each'
	/action/judges/quality-of-service/quality-of-service.rb:132:in `block in <top (required)>'
	/usr/local/bundle/gems/fbe-0.0.65/lib/fbe/regularly.rb:55:in `regularly'
	/action/judges/quality-of-service/quality-of-service.rb:30:in `<top (required)>'
	/usr/local/bundle/gems/judges-0.26.1/lib/judges/judge.rb:60:in `load'
	/usr/local/bundle/gems/judges-0.26.1/lib/judges/judge.rb:60:in `block in run'
	/usr/local/bundle/gems/elapsed-0.0.1/lib/elapsed.rb:45:in `elapsed'
	/usr/local/bundle/gems/judges-0.26.1/lib/judges/judge.rb:59:in `run'
```

### PR DESCRIPTION
**Pull request by Yegorov**
Closes #338

### CODE REVIEW

File: judges/quality-of-service/quality-of-service.rb [hunk 1]
```diff
@@ -155,8 +155,8 @@
       "repo:#{repo} type:pr is:merged closed:>#{f.since.utc.iso8601[0..9]}"
     )[:items].each do |pr|
       pr_reviews = Fbe.octo.pull_request_reviews(repo, pr[:number])
-      pr_review = pr_reviews.min_by { |r| r[:submitted_at] }
-      review_times << (pr[:merged_at] - pr_review[:submitted_at]).to_i if pr_review
+      pr_review = pr_reviews.reject { |r| r[:submitted_at].nil? }.min_by { |r| r[:submitted_at] }
```
Comment (Suban05) [inline judges/quality-of-service/quality-of-service.rb:158]: @Yegorov what do you think about the using `filter_map`? 
```rb
pr_reviews.filter_map { |r| r if r[:submitted_at] }
```

**Review by Suban05 — COMMENTED** [review]
@Yegorov all good, but I have a small question

File: judges/quality-of-service/quality-of-service.rb (hunk 1 above)
Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb:158]: @Suban05 if use `filter_map` i get Rubocop warning
```
judges/quality-of-service/quality-of-service.rb:158:30: C: [Correctable] Style/MapCompactWithConditionalBlock: Replace filter_map { ... } with select.
      pr_review = pr_reviews.filter_map { |r| r if r[:submitted_at] }.min_by { |r| r[:submitted_at] }
```
As an option, may be use `select`? What do you think?
```ruby
pr_review = pr_reviews.select { |r| r[:submitted_at] }.min_by { |r| r[:submitted_at] }
```

Comment (Suban05) [reply judges/quality-of-service/quality-of-service.rb:158]: @Yegorov good idea, let's use `select`

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb]: @Suban05 thanks, fixed

**Review by Suban05 — APPROVED** [review]

**Comment by Yegorov** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@Yegorov thanks!

**Comment by 0crat** [comment]
@Suban05 Thanks for your review! Here's your reward breakdown: +4 base points, -24 for self-review, +1.6 for 160 hits-of-code, -10 for only 5 comments, +1.25 for those comments. Final tally: 4 points (minimum). Your balance is now [+254](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the good work and aim for more comments next time! 💻👍

**Comment by 0crat** [comment]
@Yegorov Thanks for your contribution! We appreciate your effort, but there's room for improvement. You've earned +4 points, which is the minimum reward according to our [policy](https://www.yegor256.com/2014/11/14/hits-of-code.html). To maximize your bonus, consider adding more hits-of-code, ensuring code reviews, and encouraging reviewer comments. [Keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) up the good work and aim higher next time! Your running balance is [+194](https://zerocracy.github.io/judges-action/zerocracy-vitals.html).

**Comment by 0crat** [comment]
@Yegorov Thank you for your contribution! We've reviewed your work and awarded you +4 points, which aligns with our team's policy for code contributions. While we appreciate your effort, please note that we encourage more substantial contributions and thorough code reviews. Your [running balance](https://zerocracy.github.io/judges-action/zerocracy-vitals.html) is now +194. We look forward to seeing more of your work and remember to [focus on both speed and quality](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) in your future contributions.

**Pull request merged**